samza-core/src/main/scala/org/apache/samza/config/StreamConfig.scala

```diff
@@ -145,10 +145,14 @@ class StreamConfig(config: Config) extends ScalaMapConfig(config) with Logging {
     * Gets the physical name for the specified streamId.
     *
     * @param streamId             the identifier for the stream in the config.
-    * @param defaultPhysicalName  the default to use if the physical name is missing.
     * @return                     the physical identifier for the stream or the default if it is undefined.
     */
-  def getPhysicalName(streamId: String, defaultPhysicalName: String) = {
+  def getPhysicalName(streamId: String) = {
+    // Add prefix to default so the physical name is unique per job
+    val defaultPhysicalName = String.format("%s-%s-%s",
```

Comment:
Are we introducing default physical name to allow for backward compatibility ?? Or some other motivation?
Looking at jobname-jobid prefix, I don't think compatibility is the motivation. 

I also didn't realize based on this patch that we are "generating" physicalStreamName! How is this "generated" physical name used? Isn't physical name the actual name of the stream defined in the system??